standard/array.lua

```diff
@@ -435,4 +436,90 @@ function Array.min(array, compare)
 	return Array.minBy(array, function(x) return x end, compare)
 end
 
+--[[
+Finds the index of the element in an array satisfying a predicate. Returs 0
+if no element satisfies the predicate.
+
+Example:
+
+Array.findIndex({3, 5, 4, 6, 7}, function(x) return x % 2 == 0 end)
+-- returns 3
+]]
+function Array.findIndex(array, pred)
+	for ix, elem in ipairs(array) do
+		if pred(elem, ix) then
+			return ix
+		end
+	end
+	return 0
+end
+
+--[[
+Returns the unique element in an array. Returns nil if there is more than one
+distinct element, or if the array is empty.
+
+Example:
+
+Array.uniqueElement({4, 4, 4})
+-- Returns 4
+]]
+function Array.uniqueElement(elems)
+	local uniqueElem
+	for i, elem in ipairs(elems) do
+		if i ~= 1 and elem ~= uniqueElem then
+			return nil
+		end
+		uniqueElem = elem
+	end
+	return uniqueElem
+end
+
+--[[
+Groups adjacent elements of an array based on applying a transformation to the
+elements. The function returns an array of groups.
+
+The optional equals parameter specifies the equality relation of the
+transformed elements.
+
+Example:
+Array.groupAdjacentBy({2, 3, 5, 7, 14, 16}, function(x) return x % 2 end)
+-- returns {{2}, {3, 5, 7}, {14, 16}}
+]]
+function Array.groupAdjacentBy(array, f, equals)
```

Comment:
Not convinced we need this function

Reply:
Considering we already have the functions (in [array_ext.lua](https://github.com/Liquipedia/Lua-Modules/blob/main/standard/array_ext.lua))... 
It feel unnecessary to me to have both `Module:Array` and `Module:Array/Ext`?

Reply:
Discussed offline. Will be combining.

Reply:
Ok